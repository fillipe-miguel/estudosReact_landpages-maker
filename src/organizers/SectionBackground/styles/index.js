import styled, { css } from 'styled-components';

export const SectionBackgroundRoot = styled.div`
  min-height: 100vh;
  display: flex;
  align-items: center;

  ${({ theme, ...props }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${props.setBackground &&
    css`
      background-color: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    `}
  `}
`;

export default SectionBackgroundRoot;
