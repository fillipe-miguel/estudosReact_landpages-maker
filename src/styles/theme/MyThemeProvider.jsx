import P from 'prop-types';
import React from 'react';
// Styles
import { ThemeProvider } from 'styled-components';
import GlobalStyles from '../global-styles';
import theme from './theme';

function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
}

MyThemeProvider.propTypes = {
  children: P.node.isRequired,
};

export default MyThemeProvider;
