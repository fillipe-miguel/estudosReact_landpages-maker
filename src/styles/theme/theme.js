const theme = {
  // Your theme HERE
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#DC143C',

    lightBg: '#FFF',
    darkBg: '#0A1128',

    white: '#FFF',
    black: '#000',
  },
  fonts: {
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4rem',
      xxlarge: '4.8rem',
      // Pulei um pois achei que n√£o usaria !
      xxxlarge: '6.4rem',
    },
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4rem',
    xxlarge: '4.8rem',
  },
  medias: {
    lteMedium: '(max-width: 768px)',
  },
};

export default theme;
