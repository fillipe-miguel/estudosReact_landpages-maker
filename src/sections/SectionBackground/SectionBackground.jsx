import P from 'prop-types';
import React from 'react';

import SectionContainer from '../SectionContainer';
import * as Styled from './styles';

function SectionBackground({ children, setBackground }) {
  return (
    <Styled.SectionBackgroundRoot setBackground={setBackground}>
      <SectionContainer>{children}</SectionContainer>
    </Styled.SectionBackgroundRoot>
  );
}

SectionBackground.defaultProps = {
  setBackground: false,
};

SectionBackground.propTypes = {
  children: P.node.isRequired,
  setBackground: P.bool,
};

export default SectionBackground;
