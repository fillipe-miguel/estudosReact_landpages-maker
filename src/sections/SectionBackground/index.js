import SectionBackground from './SectionBackground';

export default SectionBackground;
