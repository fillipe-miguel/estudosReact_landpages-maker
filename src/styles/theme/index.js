import MyThemeProvider from './MyThemeProvider';
import theme from './theme';

export { MyThemeProvider, theme };
