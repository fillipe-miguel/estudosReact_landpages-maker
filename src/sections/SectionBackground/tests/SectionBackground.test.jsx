/* eslint-disable no-undef */
import { render } from '@testing-library/react';
// Styles
import MyThemeProvider from '../../../styles/theme/MyThemeProvider';
import theme from '../../../styles/theme/theme';
// Component
import SectionBackground from '..';

describe('<SectionBackground />', () => {
  // ================================= Teste 01 =================================
  it('should render with bg darker', () => {
    const { container } = render(
      <MyThemeProvider>
        <SectionBackground setBackground>
          <h1>Teste</h1>
        </SectionBackground>
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.primaryColor,
    });
  });

  // ================================= Teste 01 =================================
  it('should render with bg light', () => {
    const { container } = render(
      <MyThemeProvider>
        <SectionBackground>
          <h1>Teste</h1>
        </SectionBackground>
      </MyThemeProvider>,
    );

    expect(container.firstChild).toHaveStyle({
      'background-color': theme.colors.white,
    });
  });

  // ================================= Teste 01 =================================
  it('should match to snapshot', () => {
    const { container } = render(
      <MyThemeProvider>
        <SectionBackground>
          <h1>Teste</h1>
        </SectionBackground>
      </MyThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
