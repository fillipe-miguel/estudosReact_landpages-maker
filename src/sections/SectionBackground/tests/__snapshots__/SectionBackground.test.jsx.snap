// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should match to snapshot 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  width: 100%;
  padding: 3.2rem;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFF;
  color: #0A1128;
}

@media (max-width:768px) {
  .c1 {
    padding-inline: 3.2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h1>
      Teste
    </h1>
  </div>
</div>
`;
